awscli/customizations/s3/subcommands.py

```diff
@@ -266,12 +266,20 @@ def get_endpoint(service, region, endpoint_url, verify):
                                 verify=verify)
 
 
+def get_client(session, region, endpoint_url, verify):
+    return session.create_client('s3', region_name=region,
+                                 endpoint_url=endpoint_url, verify=verify)
+
+
 class S3Command(BasicCommand):
     def _run_main(self, parsed_args, parsed_globals):
         self.service = self._session.get_service('s3')
         self.endpoint = get_endpoint(self.service, parsed_globals.region,
                                      parsed_globals.endpoint_url,
                                      parsed_globals.verify_ssl)
+        self.client = get_client(self._session, parsed_globals.region,
```

Comment:
Why is this is the S3Command now?  Only the ListCommand uses it.  I think once we've moved more commands over it makes sense to move this over, but I'd rather keep these changes minimal.